### DISCUSSION TITLE
Popup, reasons why methods addClass/ removeClass are not returning the popup instance

### DISCUSSION THREAD

**Discussion by ferdicus** [maplibre/maplibre-gl-js#3972] [Q&A]
I've noticed, that some Popup methods do return the instance, while others don't (like [`addClass`](https://github.com/maplibre/maplibre-gl-js/blob/27d9c86b1/src/ui/popup.ts#L481-L485) for example).

Is there a specific rationale behind this, or is it simply an omission? 

This makes chaining of methods possible in this way
```js
      new Popup({
        closeButton: false,
        closeOnClick: false,
        anchor: 'bottom',
      })
        .setLngLat([myLng, myLat])
        .setHTML('<div>foobar</div>')
        .addTo(map)
        .addClassName('my-class');
```

but not in this way
```js
      new Popup({
        closeButton: false,
        closeOnClick: false,
        anchor: 'bottom',
      })
        .setLngLat([myLng, myLat])
        .addClassName('my-class')
        .setHTML('<div>foobar</div>')
        .addTo(map);
```

thanks in advance

**Comment by HarelM** [comment]
Probably a mistake, feel free to open a PR.
